Avoid duplicate calls to getParameterTypes and getGenericParameterTypes

Postremus opened the pull request:
Calling getParameterTypes or getGenericParameterTypes creates clones of the underlying arrays.
Doing so during iterations is costly, so lets better not do that.

This again saves saves a few ms. I saw both of these calls at "0,8% of all" according to the flamegraphs.

Related to #21552

Postremus commented:
Will have to rebase these changes, apperantly more was included in the commit then I originally intended.

Postremus commented:
This is now ready for review.

gsmet commented on independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/spi/ResteasyReactiveResourceInfo.java:
```diff
@@ -102,8 +103,10 @@ public Method getResourceMethod() {
     public Annotation[] getParameterAnnotations(int index) {
-        // Should we cache this?
-        return getMethod().getParameterAnnotations()[index];
+        if (parameterAnnotations == null) {
```
Given how `getMethod()` looks in the same class, there's a good chance this might need a synchronization pattern.

geoand replied:
Yeah, I'd say let's just skip the change to this method.

gsmet replied:
@Postremus could you revert this specific change and squash? After that I think it's good to go.

Postremus replied:
@gsmet I will do a squash & rebase this evening (CEST).

Postremus replied:
done

gsmet reviewed (changes requested):
Overall, this looks very good. I added a small comment about synchronization in a RESTEasy Reactive class. @geoand would know more for sure.

Postremus committed:
Avoid duplicate calls to getParameterTypes and getGenericParameterTypes
Calling getParameterTypes or getGenericParameterTypes creates clones of the underlying arrays.
Doing so during iterations is costly, so lets better not do that.
